packages/webapp/src/webview/ui/components/GuidedAnswerNode/Middle/Middle.tsx

```diff
@@ -23,63 +23,82 @@ let firstTimeFocus = true;
  * @returns - The middle react element
  */
 export function Middle(props: {
-    activeNode: GuidedAnswerNodeType;
+    activeNode: GuidedAnswerNode;
     enhancedBody: ReactElement | undefined | null;
 }): ReactElement {
     const appState = useSelector<AppState, AppState>((state) => state);
     firstTimeFocus = true;
-    return appState.guideFeedback === false ? (
-        <NotSolvedMessage />
-    ) : (
-        <div id="middle">
-            <div className="body_container">
-                <header>{props.activeNode.TITLE}</header>
-                <div id="hr"></div>
-                {props.enhancedBody ? (
-                    <FocusZone direction={FocusZoneDirection.vertical} isCircularNavigation={true}>
-                        {props.enhancedBody}
-                    </FocusZone>
-                ) : (
-                    <FocusZone direction={FocusZoneDirection.vertical} isCircularNavigation={true}>
-                        <div className="content" dangerouslySetInnerHTML={{ __html: props.activeNode.BODY }}></div>
-                    </FocusZone>
+
+    useEffect(() => {
```

Comment:
What is the reason for `useEffect` here?

Why not just 

```typescript 
if (appState.activeGuidedAnswer) {
            actions.updateLastVisitedGuide([
                ...appState.lastVisitedGuides,
                {
                    tree: appState.activeGuidedAnswer,
                    nodePath: appState.activeGuidedAnswerNode,
                    createdAt: new Date().toISOString()
                }
            ]);
        }
```

Reply:
because the component is being re-rendered each time state changes, and we don't want to add a last visited entry each time, so with useEffect I add an entry only when the node changes